src/pynxtools/nexus/nexus.py

```diff
@@ -31,13 +32,93 @@
 )
 
 
+def decode_string(
+    string_obj: Union[np.ndarray, bytes, str], decode: bool = True
+) -> Optional[Union[str, List[str], np.ndarray, bytes]]:
+    """
+    Decodes a numpy ndarray of byte objects to a Python string or list of strings.
+    Also handles single bytes and string objects. If `decode` is False, returns the
+    initial value without decoding.
+
+    Args:
+        string_obj: A numpy ndarray, bytes, or str that may need decoding.
+        decode: A boolean flag indicating whether to perform decoding.
+
+    Returns:
+        A decoded string, a list of decoded strings, or the initial value if `decode` is False.
+        Returns None if the input is empty or invalid.
+
+    Raises:
+        ValueError: If `string_obj` is not one of the supported types or decoding fails.
+    """
+    if not decode:
+        # Return the initial value without decoding
+        if not isinstance(string_obj, (np.ndarray, bytes, str)):
+            raise ValueError(
+                f"Unsupported type {type(string_obj)}. Expected np.ndarray, bytes, or str."
+            )
```

Comment:
This does not work simply like this, because I fed every value through the decode function before, i.e. also numerical ones.

Reply:
removed the type checking, now the decoding only happens on
- bytes objects -> returns list
- list of bytes or mixed bytes/str -> returns list of str
- np.arrays of of bytes or mixed bytes/str -> returns np.arrays of str